### PR TITLE
quilt-bot: Change reviewer assignment

### PR DESCRIPTION
**Pull request by kayousterhout**
This commit changes the reviewer assignment to a new scheme,
where every PR first gets assigned a reviewer from the entire
Quilt team, and then once that review is completed, if it
was from a non-committer, quilt-bot assigns a second-round
reviewer who is a committer.

### CODE REVIEW

File: review.go [hunk 1]
```diff
@@ -40,46 +40,66 @@ var committerIndex = rand.Intn(100)
...
 
 	var assignee string
-	if byCommitter {
+	if len(reviews) == 0 {
+		// The pull request has had no reviews, so assign a reviewer.
 		assignee = chooseReviewer(members, &memberIndex, pr)
-	} else {
+	} else if nonCommitterApproved {
+		// The pull request was approved by a non-committer, so assign
+		// to a commiter to merge (and optionall do a last review).
```
Comment (kklin) [inline review.go]: s/optionall/optionally

File: review.go [hunk 2]
```diff
@@ -40,46 +40,66 @@ var committerIndex = rand.Intn(100)
...
+		for _, committer := range committers {
+			if committer == *review.User.Login {
+				return
+			}
+		}
+		if review.State == "APPROVED" {
+			if prByCommitter {
+				// PRs by committers only need one review, so nothing
+				// else needs to be assigned.
+				return
```
Comment (kklin) [inline review.go]: These `return`s make me a bit nervous. It made it hard for me to tell whether the `if` statement below actually accounted for all the cases. I'm not sure it's worth the effort to consolidate the logic that decides how to assign a reviewer though given that this code isn't all that important.

Comment (kklin) [reply review.go]: I think it'd be worth doing if you had an idea for a simple way to make the if statement easier to follow though.

Comment (kayousterhout) [reply review.go]: Ok I tried a new version (which fixes a minor issue too).  Let me know what you think.  If you like it better I'll squash the commits.

Comment (kklin) [reply review.go]: Eh, I think what you had before is easier to follow. Maybe our review rules are just too complicated.

File: review.go [hunk 3]
```diff
@@ -37,63 +37,110 @@ func runReview(client *github.Client) {
 var memberIndex = rand.Intn(100)
 var committerIndex = rand.Intn(100)
 
+func userInList(user *string, listOfUsers *[]string) bool {
```
Comment (kklin) [inline review.go]: Why take a pointer to the listOfUsers?

File: review.go [hunk 4]
```diff
@@ -37,63 +37,110 @@ func runReview(client *github.Client) {
...
+		assignReviewer(client, pr, &committers, &committerIndex)
 	}
+	// Either there's an in-process review (e.g., a non-committer has done
+	// a review but not approved it yet), in which case we don't need to
+	// assign anyone else yet, or a committer has seen the PR, so no one
+	// else needs to review it.
+}
+
+func assignReviewer(client *github.Client, pr *github.PullRequest,
+	reviewerOptions *[]string, index *int) {
```
Comment (kklin) [inline review.go]: Why is reviewerOptions a pointer?

**Pull request merged**